pkg/registry/core/serviceaccount/storage/token.go

```diff
@@ -146,8 +153,31 @@ func (r *TokenREST) Create(ctx context.Context, name string, obj runtime.Object,
 				return nil, errors.NewBadRequest(fmt.Sprintf("cannot bind token for serviceaccount %q to pod running with different serviceaccount name.", name))
 			}
 			uid = pod.UID
+			if utilfeature.DefaultFeatureGate.Enabled(features.ServiceAccountTokenPodNodeInfo) {
+				if nodeName := pod.Spec.NodeName; nodeName != "" {
+					newCtx := newContext(ctx, "nodes", nodeName, api.SchemeGroupVersion.WithKind("Node"), true)
+					// set ResourceVersion=0 to allow this to be read/served from the apiservers watch cache
+					nodeObj, err := r.nodes.Get(newCtx, nodeName, &metav1.GetOptions{ResourceVersion: "0"})
+					if err != nil {
+						klog.V(4).ErrorS(err, "failed fetching node pod is assigned to", "pod", klog.KObj(pod), "podUID", pod.UID)
```

Comment:
if we can't look up the node, I wonder if we should still add the node name claim and uid claim and set the uid claim to "" ... consistently having those claims available for pods with spec.nodeName set saves callers looking up the pod, still records the node the pod was associated with at the time the token was created so anyone doing enforcement on requests made with the token can still do so based on the node name, at least

@enj, thoughts on this?

Reply:
This seems reasonable to me, and is certainly helpful given this is informational only anyway.

Are there any concerns around how we communicate this sort of info to users? We can advise anyone checking `node` claims to always verify the uid is set if they need an integrity protected value.

This extends to TokenReview too, as TokenReview will return the `name` as ExtraInfo but without the UID.

Do we need to/can we do anything to communicate this distinction more clearly to users of the authentication APIs?

Reply:
We also don't have any explicit page that documents using TokenRequest directly in the first place from what I can see? Unless there is a `kubectl` based guide I have missed (we also don't have generated reference docs for `kubectl create token` (e.g. for clusterrole https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#-em-clusterrole-em-)

Reply:
yeah, make notes in the placeholder docs PR of what we want to add

here, if we hit an error looking up the node, I guess log and set the node to `&api.Node{ObjectMeta:metav1.ObjectMeta{Name:nodeName}}`

```suggestion
						klog.V(4).ErrorS(err, "failed fetching node for pod", "pod", klog.KObj(pod), "podUID", pod.UID, "nodeName", nodeName)
```